converter/shieldhit/parser.py

```diff
@@ -207,6 +218,34 @@ def _get_detector_by_uuid(self, detect_uuid: str) -> Optional[str]:
 
         raise ValueError(f"No detector with uuid {detect_uuid}")
 
+    def _parse_quantity_settings(self, quantity_dict: dict) -> dict or None:
+        """Parses settings from the input json into the quantity settings property"""
+
+        def createNameFromSettings() -> str:
+            return f"""{'Absolute_' 
+                if 'primaries' in quantity_dict 
+                else 'Rescaled_' 
+                if 'rescale' in quantity_dict 
+                else ''}{quantity_dict['name']}{('_to_'+quantity_dict['medium']) 
+                if 'medium' in quantity_dict 
+                else ('_to_'+self._get_material_by_uuid(quantity_dict['materialUuid']).name) 
+                if 'materialUuid' in quantity_dict 
+                else ''
+            }"""
+
+        if all(map(lambda el: el not in quantity_dict, ['medium', 'offset', 'primaries', 'materialUuid', 'rescale'])):
+            return None
+
+        return QuantitySettings(
+            name=createNameFromSettings(),
+            medium=quantity_dict["medium"] if 'medium' in quantity_dict else None,
```

Comment:
This should be equivalent to following:

```suggestion
            medium=quantity_dict.get("medium"),
```

if "medium" key is missing, `.get` would return None